### ISSUE TITLE
New warning with Gradle 7.1

### ISSUE THREAD

**Issue by pvegh** [pinterest/ktlint#1161]
I'm getting this warning with Gradle 7.1 RC1:
```
- Gradle detected a problem with the following location: '.../foo/build/intermediates/ktLint/runKtlintFormatOverTestSourceSet_errors.bin'. Reason: Task ':foo:ktlintTestSourceSetFormat' uses this output of task ':foo:runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. 
Please refer to https://docs.gradle.org/7.1-rc-1/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

**Comment by pvegh** [comment]
My bad, it must be a problem on the gradle plugin side.
